faq.md

```diff
@@ -163,15 +170,21 @@ Explicit returns are only used if an implicit return is impossible because you a
 ### Why aren't function signatures inferred?
 </a>
 
-- Mandatory function signatures help enforce interface stability at both the module and crate level.
-- It improves code comprehension for the programmer, eliminating the need for an IDE running an inference algorithm across an entire crate to be able to guess at a function's argument types; it's always explicit and nearby.
+In Rust, declarations tend to come with explicit types, while actual code has its types inferred. There are several reasons for this design:
+
+- Mandatory declaration signatures help enforce interface stability at both the module and crate level.
+- Signatures improve code comprehension for the programmer, eliminating the need for an IDE running an inference algorithm across an entire crate to be able to guess at a function's argument types; it's always explicit and nearby.
 - Mechanically, it simplifies the inference algorithm, as inference only requires looking at one function at a time.
 
 <a href="#why-does-match-have-to-be-exhaustive" name="why-does-match-have-to-be-exhaustive">
 ### Why does `match` have to be exhaustive?
 </a>
 
-`match` being exhaustive has some useful properties. First, if every possibility is covered by the `match`, adding variants to the `enum` in the future will cause a compilation failure, rather than an error at runtime. Second, it makes the semantics of the default case explicit: in general, the only safe way to have a non-exhaustive `match` would be to panic the thread if nothing is matched. Early versions of Rust did not require `match` cases to be exhaustive and it was found to be a great source of bugs.
+To aid in refactoring and clarity.
+
+First, if every possibility is covered by the `match`, adding variants to the `enum` in the future will cause a compilation failure, rather than an error at runtime. This compiler assistance makes fearless refactoring possible in Rust.
```

Comment:
"This compiler assistance makes fearless refactoring possible in Rust" is pretty strong. This is a single factor that makes refactoring easy in Rust.